### PR TITLE
I/O API refactoring

### PR DESCRIPTION
**Pull request by totakke**
Breaking refactoring of I/O API.

Old `cljam.io` is not user-friendly, moving to `cljam.io.protocols`. Basically, this namespace should not be used directly on other projects. It can be used in cljam project or for advanced usage.

Public namespaces related to a format and its binary equivalent are integrated. For example,

* `cljam.io.sam` supports SAM and BAM,
* `cljam.io.vcf` supports VCF and BCF,
* `cljam.io.sequence` supports FASTA and TwoBit.

These have read/write functions wrapping `cljam.io.protocols`. Only commonly-used functions are wrapped. A user just `require` a single namespace that handles a format.

```clojure
(require '[cljam.io.sam :as sam])

(with-open [rdr (sam/reader "foo.bam")]
  (sam/read-alignments rdr))
```

### CODE REVIEW

File: src/cljam/io/util.clj [hunk 1]
```diff
@@ -136,38 +140,3 @@
     #"(?i)\.bcf$" :bcf
     #"(?i)\.bed" :bed
     (throw (IllegalArgumentException. "Invalid file type"))))
-
-(defn ^Closeable reader
```
Comment (alumi) [inline src/cljam/io/util.clj]: How about leaving `reader/writer` functions here?
It's true that they're unnecessary for developing apps, but I think those versatile functions are still useful for REPL.

Comment (totakke) [inline src/cljam/io/util.clj]: That causes cyclic load. `cljam.sam` loads `cljam.io.util` for `cljam.sam/reader`, and `cljam.io.util` loads `cljam.sam` for `cljam.io.util/reader`. If `reader/writer` are realy necessary, I have to reconsider namespace allocation.

Comment (alumi) [inline src/cljam/io/util.clj]: Thank you for pointing that out. OK, Let's remove them.

**Comment by alumi** [comment]
Merged. Thank you! 👍

**Comment by totakke** [comment]
Thank you for review!

**Pull request merged**